vulnerabilities/helpers.py

```diff
@@ -267,3 +267,81 @@ def _get_gh_response(gh_token, graphql_query):
     endpoint = "https://api.github.com/graphql"
     headers = {"Authorization": f"bearer {gh_token}"}
     return requests.post(endpoint, headers=headers, json=graphql_query).json()
+
+
+def get_reference_id(url: str):
+    """
+    Return the reference id from a URL
+    For example:
+    >>> get_reference_id("https://github.com/advisories/GHSA-c9hw-wf7x-jp9j")
+    'GHSA-c9hw-wf7x-jp9j'
+    """
+    url_parts = url.split("/")
+    last_url_part = url_parts[-1]
+    return last_url_part
+
+
+def extract_references(urls: List[str]):
+
+    # To avoid circular import
```

Comment:
IMHO this means this is likely best as a factory class method of Reference....
may be more of less:
```Python
class Reference:
    @classmethod
    def from_url(cls, url):
        reference_id = get_reference_id(url)
        return cls(.....)
```